src/main/kotlin/com/sourcegraph/cody/chat/ui/ChatPanel.kt

```diff
@@ -80,11 +80,15 @@ class ChatPanel(
 
   @RequiresEdt
   fun addOrUpdateMessage(message: ChatMessage, index: Int) {
-    if (messagesPanel.componentCount == 1) {
+    val numberOfMessagesBeforeAddOrUpdate = messagesPanel.componentCount
+    if (numberOfMessagesBeforeAddOrUpdate == 1) {
       llmDropdown.updateAfterFirstMessage()
       promptPanel.updateEmptyTextAfterFirstMessage()
     }
     messagesPanel.addOrUpdateMessage(message, index)
+    if (numberOfMessagesBeforeAddOrUpdate < messagesPanel.componentCount) {
```

Comment:
In which situation that if may not evaluate to `true`?

Reply:
For message update